File: bug-zoo/species/BZ-SHAPE-007-csharp-null-boundary-equivalence/dropped/csharp-native/library/src/UserDirectory.cs
namespace BugZoo.CSharpNullBoundary;

public static class UserDirectory
{
    public static string Lookup(string? name)
    {
        if (name is null)
        {
            throw new ArgumentNullException(nameof(name), "name must be non-null");
        }

        return "user:" + name.ToUpperInvariant();
    }
}

File: menagerie/bug-zoo/species/BZ-SHAPE-005-null-boundary-equivalence/java/fixed/spring-web/harness/src/main/java/zoo/UserDirectory.java
package zoo;

import org.springframework.web.bind.annotation.RequestParam;

public final class UserDirectory {
    public String lookup(@RequestParam String name) {
        if (name == null) {
            throw new IllegalArgumentException("name must be non-null");
        }
        return "user:" + name.toUpperCase();
    }
}
